stm32-modules/thermocycler-gen2/firmware/thermal/freertos_thermal_plate_task.cpp

```diff
@@ -82,7 +82,7 @@ static auto read_thermistor(const ADCPinMap &pin) -> uint16_t {
     auto result =
         _adc.at(static_cast<uint8_t>(pin.adc_index)).read(pin.adc_pin);
     if (std::holds_alternative<ADS1115::Error>(result)) {
-        return 0;
+        return 0xFFFF - (uint16_t)std::get<ADS1115::Error>(result);
```

Comment:
What's your thought process on this one? Why not just the cast?

Reply:
This was in order to disambiguate an error response from just a faulty reading. I think that just using a cast and updating the error enumeration to start at 1 would be more appropriate.